components/brave_wallet/browser/internal/hd_key.cc

```diff
@@ -136,19 +133,10 @@ std::unique_ptr<HDKey> HDKey::GenerateFromSeed(base::span<const uint8_t> seed) {
     return nullptr;
   }
 
-  SecureVector hmac(kSHA512Length);
-  unsigned int out_len;
-  if (!HMAC(EVP_sha512(), kMasterSecret, sizeof(kMasterSecret), seed.data(),
-            seed.size(), hmac.data(), &out_len)) {
-    LOG(ERROR) << __func__ << ": HMAC_SHA512 failed";
-    return nullptr;
-  }
-  DCHECK(out_len == kSHA512Length);
+  auto hmac = HmacSha512(base::byte_span_from_cstring(kMasterSecret), seed);
```

Comment:
Should this be typed as a SecureVector still?

Reply:
Not sure why it was a secure vector before. `seed` and `mnemonic` are not 'secure' in this way  but have same or higher level of importance.

It makes sense to clear private keys from memory when wallet is locked after browser being idle for some time. But I don't see a consistent approach when/what we should clear at active runtime.

Reply:
Looks like this is when it was introduced - https://github.com/brave/brave-core/pull/16792

It seems like we may have just missed some of these as refactors occurred since the change. Let's keep this as a SecureVector in here and then can do a follow-up issue to be more consistent about using SecureVector once ZCash work is finished up. In general, my preference is that if it's a secret we don't want left in memory after the wallet is locked then we should set it as a SecureVector. The one area where this might be a bit tricky is going to be zcash with their various key structures hence wanting to wait until that's finished.